### PR TITLE
Add project alerting list and detail pages

### PR DESCRIPTION
**Pull request by catherineluse**
This PR adds:

- an AlertmanagerConfig list page
- an AlertmanagerConfig detail page that has a receivers tab and a route tab
- the receivers tab has a list of receivers
- there is a detail page and separate create/edit form for receivers
- It adds props to ActionMenu so that it can be used without a dependency on Vuex, which is needed for forms that aren't for Kubernetes resources, like the receiver forms.

Here are the things that were on my to-do list, but didn't make it into this PR:

Top 3 biggest items:
- In the receiver list view, the context menus on each table row still need "Edit Config" and "Delete" menu items. I was about 70 percent through adding that functionality and that work in progress is in this separate PR over here. https://github.com/catherineluse/dashboard/pull/2
- In the AlertmanagerConfig detail page, on the route tab, the top field needs to be a multiselect where you can select receiver names. (names of receivers in the same AlertmanagerConfig)
- The individual receiver forms need to have secret selectors instead of plain input fields for configuring credentials. API details are here: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#receiver As discussed in standup I can pick this up on Monday.

Smaller to-do items:
- In the AlertmanagerConfig list view, in the Receivers column, add names and icons for a summary of receivers (a summary of configs across multiple receivers in the AlertmanagerConfig).
- Fix a bug where refreshing the receiver edit page makes the data appear empty.
- Add translation keys for headers - 'create receiver...'
- Add translation key for error message 'Could not find the AlertmanagerConfig resource'
- Remove the console.logs
- On the deprecated pages for the separate Route and Receiver custom resources, put a banner saying they are deprecated and you should use AlertmanagerConfigs.
- Put developer comments on the source code for the deprecated forms to say only AlertmanagerConfig receivers should be updated going forward.

### CODE REVIEW

File: components/CruResource.vue [hunk 1]
```diff
@@ -92,9 +92,11 @@ export default {
     }
   },
 
-  data() {
+  data(props) {
     const yaml = this.createResourceYaml();
 
+    console.log('done route in alertmanagerConfig ', props.doneRoute);
```
Comment (Sean-McQ) [inline components/CruResource.vue:98]: console statement

File: edit/monitoring.coreos.com.alertmanagerconfig/index.vue [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    translateReceiverTypes() {
+      return this.receiverTypes.map((receiverType) => {
+        return {
+          ...receiverType,
+          label: this.t(receiverType.label)
+        };
+      });
+    },
+    addNewReceiver(name) {
+      // const existingReceiversOfSelectedType = this.value.spec.receivers[this.newReceiverType] || []
```
Comment (Sean-McQ) [inline edit/monitoring.coreos.com.alertmanagerconfig/index.vue:98]: add ToDo: so we don't forget this

File: edit/monitoring.coreos.com.alertmanagerconfig/routeConfig.vue [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+};
+
+</script>
+<template>
+  <div>
+    <div class="row mb-20">
+      <div class="col span-6">
+        <!-- <LabeledSelect
```
Comment (Sean-McQ) [inline edit/monitoring.coreos.com.alertmanagerconfig/routeConfig.vue:46]: add a ToDo: here so we don't forget

File: edit/monitoring.coreos.com.alertmanagerconfig/types/webhook.add.vue [hunk 4]
```diff
@@ -0,0 +1,63 @@
+<script>
+import ButtonDropdown from '@/components/ButtonDropdown';
+import { _VIEW } from '@/config/query-params';
+
+export const MS_TEAMS_URL = 'http://rancher-alerting-drivers-prom2teams.cattle-monitoring-system.svc:8089/v2/connector';
```
Comment (Sean-McQ) [inline edit/monitoring.coreos.com.alertmanagerconfig/types/webhook.add.vue:5]: magic strings, we do this all over the place though but we should eventually move these into an env var somewhere...

File: pages/c/_cluster/monitoring/alertmanagerconfig/_alertmanagerconfigid/receiver.vue [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+        return EDITOR_MODES.VIEW_CODE;
+      } else if ( this.showPreview ) {
+        return EDITOR_MODES.DIFF_CODE;
+      }
+
+      return EDITOR_MODES.EDIT_CODE;
+    },
+    doneLocationOverride() {
+      // Go to AlertmanagerConfig detail page
+      // https://192.168.0.18:8005/c/c-m-9ldb2hdm/monitoring/monitoring.coreos.com.alertmanagerconfig/default/test-slack#receivers
```
Comment (Sean-McQ) [inline pages/c/_cluster/monitoring/alertmanagerconfig/_alertmanagerconfigid/receiver.vue:114]: these probably shouldn't be in here

**Review by Sean-McQ — APPROVED** [review]
Looks good to me! Few comments that I did add seemed non-functional in nature and were pretty nitpicky.

File: config/types.js [hunk 6]
```diff
@@ -103,23 +103,26 @@ export const HELM = { PROJECTHELMCHART: 'helm.cattle.io.projecthelmchart' };
 export const MONITORING = {
   ALERTMANAGER:       'monitoring.coreos.com.alertmanager',
   ALERTMANAGERCONFIG: 'monitoring.coreos.com.alertmanagerconfig',
+  // check for required values
```
[the remark this answers is not in the document]
Comment (catherineluse) [reply config/types.js:106]: I don't remember. It's removed now though

File: edit/monitoring.coreos.com.alertmanagerconfig/index.vue (hunk 2 above)
Comment (catherineluse) [reply edit/monitoring.coreos.com.alertmanagerconfig/index.vue:98]: I deleted it - it turned out to not be necessary.

File: edit/monitoring.coreos.com.alertmanagerconfig/types/webhook.add.vue (hunk 4 above)
Comment (catherineluse) [reply edit/monitoring.coreos.com.alertmanagerconfig/types/webhook.add.vue:5]: I had copied these from the older form for adding a webhook config. I'm not quite sure what to do with them, so I opened a separate issue about moving the URLs. https://github.com/rancher/dashboard/issues/5713

File: pages/c/_cluster/monitoring/alertmanagerconfig/_alertmanagerconfigid/receiver.vue (hunk 5 above)
Comment (catherineluse) [reply pages/c/_cluster/monitoring/alertmanagerconfig/_alertmanagerconfigid/receiver.vue:114]: Moving them to the AlertmanagerConfig class model instead

**Pull request merged**